.github/stale.yml

```diff
@@ -0,0 +1,57 @@
+# Configuration for probot-stale - https://github.com/probot/stale
+
+# Number of days of inactivity before an Issue or Pull Request becomes stale
+daysUntilStale: 180 # 6 months
+
+# Number of days of inactivity before an Issue or Pull Request with the stale label is closed.
+# Set to false to disable. If disabled, issues still need to be closed manually, but will remain marked as stale.
+daysUntilClose: 7
```

Comment:
Not sure if this can be done by the bot. Sometimes additional stuff needs to be done, like marking "up for grabs" or closing a related issue as well, etc ...

Reply:
It might be easier to implement these features in Drahtbot, do you plan to open-source it?

Reply:
It would also be nice if Drahtbot could automatically add to a PR the labels "Concept ACK", "Approach ACK" and even count current ACKs, that way it's easy to quickly glance at the state of a PR, or even filter PRs for reviewers. I believe @JeremyRubin had a similar idea.

Reply:
Yeah, I wanted to do that for a long time as well. Though, @DrahtBot is currently written such that it needs to scan every comment, so it will run into the rate limit quite fast.

Reply:
> it will run into the rate limit quite fast

This could be fixed by using webhooks no? When a comment/review is added to a PR Drahbot could fetch the PR's comments and apply labels. Fetching comments is limited to 100 per req, which is probably fine for most PRs and could also be cached.
And another webhook for when a user pushes to a PR, Drahbot would invalidate ACKs by removing the label.

The rate-limit doesn't seem too restrictive either? Unless I'm missing something
> User-to-server requests are limited to 5,000 requests per hour and per authenticated user

I'd love to work on this but the code is nowhere to be found :frog:.

Reply:
I am not sure how reliably webhooks are. If they are down or otherwise missed, it wouldn't be possible to catch up, unless you have  script to catch up, in which case you can just use that script all the time.

Reply:
If a webhook is missed, which I believe is relatively rare, the bot will catch up on the next webhook received (the next comment/review/push), because it will rescan the PRs comments. I don't think it's that big of a deal to miss an event, this feature is to provide heuristics anyway.

Reply:
The source code is of low quality, but I just made it public. See for example https://github.com/MarcoFalke/DrahtBot/blob/main/scripts/stale.py